### DISCUSSION TITLE
get collection id from modelQuery

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2228] [Q&A] [answered]
How to get Collection ID when using `ModelQuery()`?

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what do you mean. Could you elaborate a little more?

**Reply by Oggy4th** [reply]
Hello, there @ganigeorgiev 

Thanks for your reply

How to find out collection Id when using below code?

```
func BranchQuery(dao *daos.Dao) *dbx.SelectQuery {
	return dao.ModelQuery(&Branch{})
}

func FindBranchById(dao *daos.Dao, id string) (*Branch, error) {
	branch := &Branch{}

	err := BranchQuery(dao).AndWhere(dbx.HashExp{"id": id}).Limit(1).One(branch)

	if err != nil {
		return nil, err
	}

	return branch, nil
}
```

**Reply by ganigeorgiev** [reply, marked as answer]
I assume that you are using [Custom models](https://pocketbase.io/docs/custom-models/). The collection data is not part of the record model table (in your case that would be `Branch.TableName()`).

You can retrieve the related `Branch` collection using the [collection Dao helpers](https://pocketbase.io/docs/collection-methods/), for example:

```go
collection, err := app.Dao().FindCollectionByNameOrId("branch")
```

I don't understand your use case but if you want the collection data to be part of the `Branch` struct you can perform a join with the `_collections` table, something like:

```go
BranchQuery(dao).
  AndSelect("c.id as collectionId")
  LeftJoin("_collections as c", dbx.NewExp("c.name = 'branch'"))
  AndWhere(dbx.HashExp{"branch.id": id}). // you may have to prefix the "id" column with the branch table name
  Limit(1).
  One(branch)
```

But please note that there is not enough context to understand what are the `Branch` struct fields and what you are trying to do in general.
